packages/astro/src/cli/check/index.ts

```diff
@@ -1,101 +1,339 @@
 /* eslint-disable no-console */
-import { AstroCheck, DiagnosticSeverity } from '@astrojs/language-server';
+import { AstroCheck, DiagnosticSeverity, GetDiagnosticsResult } from '@astrojs/language-server';
 import type { AstroSettings } from '../../@types/astro';
 import type { LogOptions } from '../../core/logger/core.js';
-
 import glob from 'fast-glob';
 import * as fs from 'fs';
 import { bold, dim, red, yellow } from 'kleur/colors';
 import { createRequire } from 'module';
 import ora from 'ora';
 import { fileURLToPath, pathToFileURL } from 'url';
 import { printDiagnostic } from './print.js';
+import type { Arguments as Flags } from 'yargs-parser';
+import { debug, info } from '../../core/logger/core.js';
+import { createServer, ViteDevServer } from 'vite';
+import { createVite } from '../../core/create-vite.js';
+import type { CreateViteOptions } from '../../core/create-vite';
+import type { SyncOptions, ProcessExit } from '../../core/sync';
+import fsMod from 'fs';
 
-interface Result {
+type DiagnosticResult = {
 	errors: number;
 	// The language server cannot actually return any warnings at the moment, but we'll keep this here for future use
```

Comment:
This comment is actually no longer true, as it now returns warnings in some cases from the compiler